release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,240 @@
+          edition = Edition.ENTERPRISE;
+          repository = Repository.CLUSTER;
+          break;
+        case "graphql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.GRAPHQL;
+          break;
+        case "sql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.SQL;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown file name: " + path);
+      }
+
+      File file = new File(path);
+      load(edition, repository, file);
+    }
+    output();
+  }
+
+  void load(Edition edition, Repository repository, File file) throws Exception {
+    Category category = null;
+    String line;
+    Matcher matcher;
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8))) {
+      while ((line = reader.readLine()) != null) {
+        if (DEBUG != null) System.err.printf("ReadLine: %s%n", line);
+        matcher = Pattern.compile("^## *(\\p{Print}+) *$").matcher(line);
+        if (matcher.matches()) {
+          if (DEBUG != null) System.err.printf("matched: %s%n", matcher.group(1));
+          if (!SECTION_SUMMARY.equalsIgnoreCase(matcher.group(1)))
+            category = Category.getByDisplayName(matcher.group(1));
+          continue;
+        }
+
+        matcher = Pattern.compile("^ *- *(\\p{Print}+)$").matcher(line);
+        if (matcher.matches()) {
+          if (category == null)
+            throw new IllegalStateException(
+                "Missing category. Release note text: " + matcher.group(1));
+          addReleaseNote(new ReleaseNote(edition, category, repository, matcher.group(1)));
+        }
+      }
+    }
+  }
+
+  private void addReleaseNote(ReleaseNote releaseNote) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap =
+        editionMap.computeIfAbsent(releaseNote.edition, k -> new EnumMap<>(Category.class));
+
+    Map<Repository, List<ReleaseNote>> repositoryMap =
+        categoryMap.computeIfAbsent(releaseNote.category, k -> new EnumMap<>(Repository.class));
+
+    List<ReleaseNote> releaseNotesList =
+        repositoryMap.computeIfAbsent(releaseNote.repository, k -> new ArrayList<>());
+
+    if (releaseNote.edition.equals(Edition.ENTERPRISE)) {
+      Matcher matcher =
+          Pattern.compile("(.*) +(\\((#[0-9]+ *)+\\))$").matcher(releaseNote.releaseNoteText);
+      if (matcher.matches()) {
+        if (DEBUG != null)
+          System.err.printf(
+              "Matched::%s::%s grp1:%s grp2:%s%n",
+              releaseNote.category, releaseNote.repository, matcher.group(1), matcher.group(2));
+        releaseNote.releaseNoteText = matcher.group(1);
+      }
+    }
+
+    releaseNotesList.add(releaseNote);
+    repositoryMap.put(releaseNote.repository, releaseNotesList);
+    categoryMap.put(releaseNote.category, repositoryMap);
+    editionMap.put(releaseNote.edition, categoryMap);
+  }
+
+  void output() {
+    System.out.print("## Summary\n\n");
```

Comment:
Currently, I think it is enough to output to the stdout because I assume this script is used as cli script.
Even though, if you still think that we should abstract the file pointer, I'll change it.

What do you think?

Reply:
Thank you! Yes, I agree that given that the use case is clear, it is okay to keep it.